app/models/post.rb

```diff
@@ -1,4 +1,15 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
-  has_many :likes
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  def update_posts_counter
+    @users.each do |user|
+      user.update_column(:posts_counter, user.posts.count) if user.id == author_id
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] It would be awesome to make this method private. In order to do that, you can use the `after_save` Rails Active Callback to call the `update_posts_counter`, so every time you saved the posts, the counter will be updated. As mentioned, this is not mandatory, but you can consider implementing it.